test/lib/testManager.js

```diff
@@ -1,116 +1,92 @@
 const
   fs = require('fs'),
   path = require('path'),
+  readYaml = require('read-yaml'),
   Snippet = require('./snippet'),
   Logger = require('./helpers/logger'),
   {
-    getSupportedLanguages,
+    getSupportedSdks,
     getVersionPath
   } = require('./helpers/utils'),
   TestResult = require('./helpers/testResult');
 
 class TestManager {
-  constructor(basePath) {
-    if (basePath.indexOf('sdk-reference') === -1) {
-      throw new Error('Unable to find sdk-reference directory in basePath');
+  constructor(sdk, version, path) {
+    const supportedSdks = getSupportedSdks();
+    if (! supportedSdks.includes(sdk)) {
+      throw new Error(`Unknown SDK ${sdk}. Supported SDKs: ${supportedSdks.join(', ')}`);
     }
 
-    const [language, version, rest] = (basePath.split('sdk-reference/')[1] || '').split('/');
+    this.basePath = path;
 
-    if (! language) {
-      throw new Error('Unable to find language in basePath');
-    }
-
-    if (! version) {
-      throw new Error('Unable to find version in basePath');
-    }
+    const Sdk = require(`./sdk/${sdk}Sdk`);
+    this.sdk = new Sdk(version);
 
-    const supportedLanguages = getSupportedLanguages();
-    if (! supportedLanguages.includes(language)) {
-      throw new Error(`Unknown language ${language}. Supported languages: ${supportedLanguages.join(', ')}`);
-    }
-
-    this.basePath = basePath;
-    this.language = language;
-    this.version = version;
+    const Runner = require(`./runners/${this.sdk.name}Runner`);
+    this.languageRunner = new Runner(this.sdk);
```

Comment:
Do you still want this to be called `languageRunner` or is this now inherent to the specific SDK? (just asking)